EventListener/AttachRefreshTokenOnSuccessListener.php

```diff
@@ -107,9 +121,27 @@ public function attachRefreshToken(AuthenticationSuccessEvent $event): void
             $refreshToken = $this->refreshTokenGenerator->createForUserWithTtl($user, $this->ttl);
 
             $this->refreshTokenManager->save($refreshToken);
-            $data[$this->tokenParameterName] = $refreshToken->getRefreshToken();
+            $refreshTokenString = $refreshToken->getRefreshToken();
         }
 
-        $event->setData($data);
+        if (isset($this->cookieSettings['enabled']) && $this->cookieSettings['enabled']) {
+            $event->getResponse()->headers->setCookie(
+                new Cookie(
+                    $this->tokenParameterName,
+                    $refreshTokenString,
+                    time() + $this->ttl,
+                    $this->cookieSettings['path'],
+                    $this->cookieSettings['domain'],
+                    $this->cookieSettings['secure'],
+                    $this->cookieSettings['http_only'],
+                    false,
+                    $this->cookieSettings['same_site']
+                )
+            );
+        } else {
```

Comment:
Could we put an "if" as an argument to allow us to use both? How to raise this issue

https://github.com/lexik/LexikJWTAuthenticationBundle/issues/909